packages/puppeteer-core/src/common/Touchscreen.ts

```diff
@@ -0,0 +1,56 @@
+/**
```

Comment:
This should live in API. The structure is a little bit confusing :sweat_smile:

Reply:
But is has a generic type parameter, and a public method 'removeHandle'. Does that really belong in api?

Reply:
I could also just revert it if you don't like it

Reply:
We should move it directly to the abstract class as that it's purpose to hold common logic.

We can remove the generic and overwrite it if needed on the child class with `declare X: <type>`.

As for the public method we can omit it from the types and docs by marking it as internal.
```ts
/*
* @internal
*/
method(){}
```

Reply:
Also I thought that because it is API it should contain as little logic as possible

Reply:
Just a note about structure we have decided to use - we use API as common between the two protocols we support, all common implementation goes there.
That is why we use  `abstract class`, if it were meant to be only Documentation and Types the correct thing to use are `interfaces` and makes sure the implementation `implements` them. 

For the `declare` it's TypeScript feature that seems to be not that known details: 
You seem to need the generic for the CDP implementation to update the client .
You can take a look at `packages/puppeteer-core/src/cdp/HTTPRequest.ts` 
On the base class in `api` we say 
```ts
class HTTPRequest{
  _redirectChain: HTTPRequest[]
}
```
on the child class we say that 
```ts
class CdpHTTPRequest extends HTTPRequest {
  declare _redirectChain: CdpHTTPRequest[]
}
```
In short you can provide this allows you to scope down the type of property to more precise one.

Reply:
Interesting! I had never seen that.

About the structure: if api is for common functionality, then why is there a separate 'common' directory?

Reply:
Good question - I can see that is confusing - there are (mainly) historical reason why the structure exist: 
In general `api` is the extracted logic from the original that can be reused with the BiDi protocol,
while `common` are things that work that don't have special logic for depending on the protocol.